Update UI to support minimum and maximum shipping times.

jorgemd24 opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Part of pcTzPl-2qP-p2

This PR introduces a UI to manage minimum and maximum shipping times, and it updates the input to include increase and decrease buttons (like a stepper input). Additionally, we're considering showing "Same Day" instead of 0 to make it clearer for merchants.

![image](https://github.com/user-attachments/assets/a77918ce-f688-4aef-af4d-737cb5d3a9a5)


Since mixing a string like "Same Day" with the actual integer values for shipping times could make the code more complex, I opted to use the placeholder attribute to display "Same Day" instead of 0 days to keep things simple. However, I'm open to other suggestions or approaches.

For more context see this comment here: pcTzPl-2qP-p2#comment-3652

#### Other Steeper Inputs in WC

https://github.com/woocommerce/woocommerce/blob/64dcafe29b52cff2dee1d56f5e06caad09a0c470/packages/js/product-editor/src/components/number-control/number-control.tsx#L166-L222

### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Follow the steps mentioned in this PR to update the database: https://github.com/woocommerce/google-listings-and-ads/pull/2520
2. Navigate to Dashboard -> Free Listings -> Edit
3. Test the new input buttons by increasing or decreasing the values. If the value is 0, it should display "Same Day".
4. Verify that the data is being stored correctly.
5. Add and modify shipping times as needed.
6. Disconnect all your accounts.
7. Do the onboarding, and see if everything works correctly.

### Additional details:

I haven't updated the add and edit modals to use the stepper component yet to make reviewing this PR easier. Also, I want to ask for confirmation from the designers to see if we want to implement it there as well.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

>

jorgemd24 committed:
Add same day delivery constant

jorgemd24 committed:
Create stepper button

jorgemd24 committed:
Add max and min inputs

jorgemd24 committed:
Adjust styling

jorgemd24 committed:
Adjust max transit time

jorgemd24 committed:
handle max shipping time

jorgemd24 committed:
Merge branch 'update/shipping-time-controllers-to-work-with-max-time' into add/min-max-time-inputs

jorgemd24 committed:
Remove unsused constant

jorgemd24 committed:
Check if min shipping time is less than max time

jorgemd24 committed:
Refactor code

jorgemd24 committed:
Update add and edit shipping time modals to work with maxTime

jorgemd24 committed:
Skip the incomplete shipping time when onboarding

jorgemd24 committed:
Adjust width input

jorgemd24 committed:
Fix css linting

jorgemd24 committed:
update checkError tests

jorgemd24 committed:
Add CountriesTimeInput tests

jorgemd24 committed:
Add tests for shipping stepper

puntope commented on js/src/components/free-listings/configure-product-listings/checkErrors.js:
```diff
@@ -98,14 +98,24 @@ const checkErrors = (
+		shippingTimes.some( ( el ) => el.time > el.maxTime )
+	) {
+		errors.shipping_country_times = __(
+			'The minimum shipping time must not be less than the maximum shipping time.',
```
I believe this should be "The minimum shipping time must not be more than the maximum shipping time."

puntope commented on js/src/components/free-listings/configure-product-listings/checkErrors.test.js:
```diff
@@ -400,7 +401,7 @@ describe( 'checkErrors', () => {
 			} );
 
 			it( `When there are any shipping times is < 0, should not pass`, () => {
```
Can we add separate assertion for `time` and `max_time` is < 0

puntope commented:
It's handled by the error checker but I think it would be even better if we add some checks on the onChange method to verify that `time` is not bigger than `max_time`.

puntope commented:
> I haven't updated the add and edit modals to use the stepper component yet to make reviewing this PR easier. Also, I want to ask for confirmation from the designers to see if we want to implement it there as well.

Potentially this will be handled separately in other PR. And maybe not setting the stepper... But at least we want to set also the placeholder "Same day" instead of 0? 

<img width="650" alt="Screenshot 2024-09-09 at 12 08 53" src="https://github.com/user-attachments/assets/12133516-5b5b-4a52-a216-ee5416c512a5">

jorgemd24 committed:
Merge branch 'update/shipping-time-controllers-to-work-with-max-time' into add/min-max-time-inputs

jorgemd24 committed:
Update error description

jorgemd24 committed:
Add tests for min and max time

jorgemd24 committed:
fix js linting

jorgemd24 commented:
Thanks @puntope for looking at this, I've addressed your comments, could you please take another look?

Regards these points:

> It's handled by the error checker but I think it would be even better if we add some checks on the onChange method to verify that time is not bigger than max_time.

I think to keep consistency with the rest of the form, we could use this change to all fields or leave it as it is. I'm not sure what was the reason to show the validation only when clicking "continue," but if we think it's worth changing, we could explore it in a separate PR.

> Potentially this will be handled separately in other PR. And maybe not setting the stepper... But at least we want to set also the placeholder "Same day" instead of 0?

Thanks for the suggestion! I'll discuss this with the designers to see what's best for this form, for now I think we can keep it as it is and then I can adjust it.

puntope commented:
Thanks @jorgemd24 I found a new issue when setting `time`or `max_time` as "Same day"


https://github.com/user-attachments/assets/8bc60164-bb9a-4028-91f3-0514e1f4b25e

jorgemd24 committed:
Remove unnecessary statement as the schema already requires time and maxTime

jorgemd24 committed:
Fix issue with the orignal value

jorgemd24 commented:
Hi @puntope, thanks for reviewing this! The issue was linked to a change I made in the other PR, where `0` was being treated as an invalid value. I've removed that check since the schema already requires those fields. [e4686c5](https://github.com/woocommerce/google-listings-and-ads/pull/2594/commits/e4686c559b8a86c4214371a8974acf1cce565aca)

I also found a small issue with the handleBlur function and the original value, which I've fixed here: [f218b47](https://github.com/woocommerce/google-listings-and-ads/pull/2594/commits/f218b47483fb73194100987a4b6bb0607f97c2c5)

Could you please take another look? Thanks!

PS: The PHP tests failed because of this issue: pb22l9-3cx-p2#comment-3657 but they are OK locally.

jorgemd24 committed:
remove unused property

jorgemd24 committed:
replace deprecated property with size

puntope reviewed (approved):
Thx @jorgemd24 for the tweaks. Now it's working as expected.

Just one thing about the UI. I see the minus symbol in the right. When normally the standard is to place it in the left. 

<img width="523" alt="Screenshot 2024-09-16 at 18 25 47" src="https://github.com/user-attachments/assets/b7670bf6-7a7e-4afa-92f6-28f194cd8355">


I believe maybe we can discuss with UI. But not a blocker.

<img width="466" alt="Screenshot 2024-09-16 at 18 23 14" src="https://github.com/user-attachments/assets/03a070fd-422f-4735-be8f-a22e038f4d2e">

<img width="377" alt="Screenshot 2024-09-16 at 18 21 12" src="https://github.com/user-attachments/assets/dec0c9e1-ed6b-4c97-af98-dc007e6bc3b5">

jorgemd24 commented:
Thanks @puntope for the review

> Just one thing about the UI. I see the minus symbol in the right. When normally the standard is to place it in the left.

I understand your point, however, the new version of the product editor uses the minus symbol on the right, so the plan is to maintain that style for consistency.

![image](https://github.com/user-attachments/assets/826d98b3-93b1-4188-ab94-b6f17c07fb74)
